src/model.jl

```diff
@@ -528,6 +558,23 @@ function _build_model(
                 .* storage.sd_table.TerminalStoragePrice))
         JuMP.add_to_expression!(obj, storage_penalty)
     end
+    
+    # Pay for the cost of DR programs based on committed flexible demand
+    if demand_flexibility.enabled
+        
+        # cost for increasing flexible load
+        demand_response_penalty_up = JuMP.@expression(m, sum(
+            sum(bus_demand_flex_cost_up_amt * load_shift_up)))
+        JuMP.add_to_expression!(obj, demand_response_penalty_up)
```

Comment:
If we follow my suggestion from above, then this should be preceded by something like if `!isnothing(demand_flexibility.cost_up)` and would need to be followed with `end`.